docs/static/config-details.asciidoc

```diff
@@ -72,6 +72,10 @@ the higher percentage you can use.
 value to prevent the heap from resizing at runtime, which is a very costly
 process.
 
+* If you plan to use some plugins that leverages direct memory, like Beats, TCP or HTTP input,
+ set the direct memory size (-XX:MaxDirectMemorySize) accordingly to the scale of load
+ they need to sustain.
```

Comment:
> Do we expect users to know which plugins use direct memory?

No we don't expect. For the user it's difficult to understand which plugin uses direct memory because it's something like an internal implementation "detail" however this impacts directly the user itself. I think that the best we can provide is a list of the plugins that uses direct memory. The user, when doing memory sizing has to be aware of this, because could impact him. For example by default users tend to think that all the memory used by Logstash is the JavaHeap, but that's not true, the JVM could consume other percentage of memory that's not immediately evident.
Just as an example, by default JVM size the direct memory space to the same size of Java heap size (Xmx, the maximum heap) this means that if a user has an host of 8GB, he could think to be safe (avoid the kernel memory killer or avoid disk swapping) setting Java heap to 6Gb (he leave 2Gb for the OS), but that's not true. Because the JVM can potentially consume 6 (heap) + 6 (direct memory) + 30% of heap for its internal structures, so 14GB.

> Also, will a typical user know how much load a plugin will be asked to handle when they are sizing their JVM? Or is something they'll likely be tweaking after OOM problems.

Generally yes, when hits OOM he fix the size, but knowing upfront the load he can size correctly the host and the JVM to sustain that load without risking OOM or hit some performance issues due to swapping.